### PR TITLE
(chore) Upgrade carbon

### PR DESCRIPTION
**Pull request by usamaidrsk**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR upgrades the carbon version to the latest and does the following:

- Upgrade carbon version
- Removes carbon from declarations
- Update carbon styling imports and tokens
- Update jest configc, and use framework mocked tests

### CODE REVIEW

**Comment by NethmiRodrigo** [comment]
@usamaidrsk do we need  the `.idea` folder?

File: src/greeter/greeter.test.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';
-import { useConfig } from '@openmrs/esm-framework';
-import { Config } from '../config-schema';
+import { getDefaultsFromConfigSchema, useConfig } from '@openmrs/esm-framework';
```
Comment (NethmiRodrigo) [inline src/greeter/greeter.test.tsx:3]:
```suggestion
import { useConfig } from '@openmrs/esm-framework';
```

File: src/greeter/greeter.test.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+import { Config, configSchema } from '../config-schema';
```
Comment (NethmiRodrigo) [inline src/greeter/greeter.test.tsx:4]:
```suggestion
import { Config } from '../config-schema';
```

Comment (NethmiRodrigo) [reply src/greeter/greeter.test.tsx:4]: Unused imports

**Review by NethmiRodrigo — CHANGES_REQUESTED** [review]

File: src/boxes/extensions/box.scss [hunk 3]
```diff
@@ -20,4 +20,4 @@
   @include brand-01(background-color);
   color: white;
   padding: 8px;
```
Comment (NethmiRodrigo) [inline src/boxes/extensions/box.scss:22]:
```suggestion
  padding: layout.$spacing-03;
```
Minor nitpick, shall we change this to use carbon spacing, we'd have to use carbon's layout

**Comment by NethmiRodrigo** [comment]
Thanks @usamaidrsk! Some minor details,

File: src/boxes/extensions/box.scss (hunk 3 above)
Comment (ibacher) [reply src/boxes/extensions/box.scss:22]: `layout` and `spacing` are aliases of one another... It's the same underlying tokens.

**Review by NethmiRodrigo — APPROVED** [review]

**Comment by NethmiRodrigo** [comment]
Thanks @usamaidrsk! LGTM!

**Pull request merged**